src/Aspire.Dashboard/Components/Pages/IComponentWithTelemetry.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Dashboard.Telemetry;
+using Microsoft.AspNetCore.Components;
+
+namespace Aspire.Dashboard.Components.Pages;
+
+/// <summary>
+/// Interface for components that would like to opt-in to telemetry events, capturing
+/// 1) initialization
+/// 2) component properties on each render
+/// </summary>
+public interface IComponentWithTelemetry
+{
+    public Guid? InitializeCorrelation { get; set; }
+    public NavigationManager NavigationManager { get; }
+    public DashboardTelemetryService TelemetryService { get; }
+
+    public string ComponentId { get; }
+    public Dictionary<string, AspireTelemetryProperty> GetTelemetryProperties() => [];
+}
+
+public static class IComponentWithTelemetryExtensions
+{
+    public static async Task InitializeComponentTelemetryAsync(this IComponentWithTelemetry component)
+    {
+        await component.TelemetryService.InitializeAsync();
+
+        if (component.TelemetryService.IsTelemetryEnabled)
+        {
+            component.InitializeCorrelation = component.TelemetryService.PostUserTask(
+                TelemetryEventKeys.InitializeComponent,
+                TelemetryResult.Success,
+                properties: new Dictionary<string, AspireTelemetryProperty> {
+                    // Component properties
+                    { TelemetryPropertyKeys.DashboardComponentId, new AspireTelemetryProperty(component.ComponentId) }
+                });
+        }
+    }
+
+    public static void PostParametersSetTelemetry(this IComponentWithTelemetry component)
```

Comment:
I thought about this a little more. Recording an event on every parameter change could easily cause a lot of events that don't have much value. But at the same time, we want to capture when the parameters change so updated telemetry properties are sent.

How about this as a solution:

Store the last result from `component.GetTelemetryProperties()`. Then, when parameters change, check if the new properties are different from the old properties. If properties are different call `PostOperation` and update the stored properties. If properties are the same then do nothing.